### PR TITLE
[POC] Use Drizzle in connectors

### PR DESCRIPTION
**Pull request by flvndvd**

### CODE REVIEW

File: connectors/scripts/run_mysql_migrations.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { drizzle } from "drizzle-orm/node-postgres";
```
Comment (flvndvd) [inline connectors/scripts/run_mysql_migrations.ts:1]: Helper script to run missing migrations on a SQL database in production.

File: connectors/sql-migrations/0000_chubby_malcolm_colcord.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
+CREATE TABLE IF NOT EXISTS "connectors" (
```
Comment (flvndvd) [inline connectors/sql-migrations/0000_chubby_malcolm_colcord.sql:1]: Example of a migration generated by `npm run db:generate`.

File: connectors/sql-migrations/0001_perfect_multiple_man.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
+CREATE TABLE IF NOT EXISTS "confluence" (
```
Comment (flvndvd) [inline connectors/sql-migrations/0001_perfect_multiple_man.sql:1]: Example of a migration generated by `npm run db:generate`.

File: connectors/sql-migrations/meta/_journal.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (flvndvd) [inline connectors/sql-migrations/meta/_journal.json:1]: Drizzle journal of migrations.

File: connectors/src/resources/connectors.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { eq } from "drizzle-orm";
```
Comment (flvndvd) [inline connectors/src/resources/connectors.ts:1]: Illustration of a specialized file designed for database interactions, which could use either functions or classes.

File: connectors/src/resources/db/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+// Retrieve only the selected field.
+const partialResult = await db
+  .select({ createdAt: connectors.createdAt })
+  .from(connectors)
+  .where(
+    or(eq(connectors.id, 42), eq(connectors.dataSourceName, "managed-notion"))
+  );
+
+// Fetch ConfluenceConfiguration with the connector.
+const confluenceConfigurationWithConnector =
```
Comment (flvndvd) [inline connectors/src/resources/db/index.ts]: We won't support these types of relational queries. Engineers will be encouraged to use `.join()` as necessary.

File: connectors/src/resources/db/schema/confluence.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  timestamp,
+  uniqueIndex,
+  varchar,
+} from "drizzle-orm/pg-core";
+
+import { connectors } from "@connectors/resources/db/schema/connectors";
+
+// /!\ The schema files SHOULD NOT contain any runtime logic besides defining your DB schema.
+
+export const confluenceConfigurations = pgTable(
```
Comment (flvndvd) [inline connectors/src/resources/db/schema/confluence.ts]: Example definition of a database table using Drizzle.

File: connectors/src/resources/db/schema/connectors.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import {
```
Comment (flvndvd) [inline connectors/src/resources/db/schema/connectors.ts:1]: Example of a schema file definition.

File: connectors/src/resources/db/schema/github.ts [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    }).notNull(),
+    lastSeenAt: timestamp("lastSeenAt", {
+      withTimezone: true,
+      mode: "string",
+    }).notNull(),
+    repoId: varchar("repoId", { length: 255 }).notNull(),
+    repoLogin: varchar("repoLogin", { length: 255 }).notNull(),
+    repoName: varchar("repoName", { length: 255 }).notNull(),
+    sourceUrl: varchar("sourceUrl", { length: 255 }).notNull(),
+    connectorId: integer("connectorId").references(() => connectors.id, {
```
Comment (flvndvd) [inline connectors/src/resources/db/schema/github.ts:34]: Example of a foreign key definition.

File: connectors/src/resources/db/schema/schema.ts [hunk 10]
```diff
@@ -0,0 +1,1077 @@
+import {
```
Comment (flvndvd) [inline connectors/src/resources/db/schema/schema.ts:1]: Rest of the current schema to split.

**Comment by lasryaric** [comment]
I love what I am seeing here 👍

**Pull request closed without merging**